container.go

```diff
@@ -50,6 +53,20 @@ type Container struct {
 	graph.Graph
 }
 
+// InvokeOnce only allows function invocation once to register the
+// return types. If return types are already registered, specific error
+// is returned to the caller
+func (c *Container) InvokeOnce(t interface{}) error {
+	ctype := reflect.TypeOf(t)
+	if ctype.Kind() != reflect.Func {
+		return errors.Wrapf(errParamType, _invokeErr, ctype)
+	}
+	if err := c.ValidateReturnTypes(ctype, true); err != nil {
+		return ErrInvokeOnce
+	}
+	return c.Invoke(t)
```

Comment:
`InvokeOnce` should be recursive, i.e. it should InvokeOnce other things as well, otherwise this doesn't solve the problem we need addressing.

What we want to achieve is that the order of operations in fx.Start doesn't matter. In this case, we will still double-invoke things given a particular order.